a10_neutron_lbaas/vthunder/instance_manager.py

```diff
@@ -377,6 +379,34 @@ def plumb_instance_subnet(self, instance_id, subnet_id, allowed_ips, wrong_ips=[
         network_id = subnet["subnet"]["network_id"]
         return self.plumb_instance(instance_id, network_id, allowed_ips, wrong_ips=wrong_ips)
 
+    def build_server_with_defaults(self, server, vthunder_config):
+        copy_keys = [("image", "glance_image"),
+                     ("flavor", "nova_flavor"),
+                     ("username", "username"),
+                     ("password", "password"),
+                     ("api_version", "api_version"),
+                     ("port", "port"),
+                     ("protocol", "protocol")
+                     ]
+
+        resources.remove_attributes_not_specified(server)
```

Comment:
`remove_attributes_not_specified` doesn't modify the dictionary, it returns a new one